### ISSUE TITLE
Can not run version 6.0.0 or 6.1.1

### ISSUE THREAD

**Issue by gianluca1606** [oauth2-proxy/oauth2-proxy#917]
Hello guys, I followed this tutorial [https://devopsloggers.com/2020/05/30/oauth2_proxy-google-authentication-using-nginx-on-ubuntu-18-04/](url) and the only thing i did differently was replacing the version 5.1.1 with the newest version (oauth2-proxy-v6.1.1.linux-amd64.tar.gz).

I have an ubuntu 20.04 server - linux amd 64.

The error I get is that the oauth2-proxy fails to start. So i tried with the version 6.0.0 since there is a "go" version like in the tutorial (oauth2-proxy-v6.0.0.linux-amd64.go1.14.2.tar.gz.)

And I got the same error.

But with the version from the tutorial everything works fine.

Here are my configuration files:

```
[Unit]
Description=oauth2-proxy daemon service
After=syslog.target network.target

[Service]
User=www-data
Group=www-data

ExecStart=/usr/local/bin/oauth2_proxy -config=/etc/oauth2-proxy/oauth2-proxy.cfg
ExecReload=/bin/kill -HUP $MAINPID

KillMode=process
Restart=always

[Install]
WantedBy=multi-user.target
```

 ```
http_address = "127.0.0.1:4180"
 https_address = ":443"
 reverse_proxy = true
redirect_url = "https://mydomain.de/oauth2/callback"
client_id = "123456"
client_secret = "123456"
authenticated_emails_file = "/etc/oauth2-proxy/authorized_emails.txt"
cookie_name = "_oauth2_proxy"
cookie_secret = "kjsdlfjsdkjflsdjkf=="
# cookie_domains = "gmail.com"
# cookie_expire = "168h"
cookie_refresh = "1h"
cookie_secure = true
cookie_httponly = false
```

On version 6.1.1 i was able to start the server, if i commented out everything in the config file. 

What I am missing?

**Comment by JoelSpeed** [comment]
Please share the logs from the OAuth2 Proxy process. It will explain in the logs why it is unable to start

**Comment by gianluca1606** [comment]
> Please share the logs from the OAuth2 Proxy process. It will explain in the logs why it is unable to start

I do not now how to get the logs but this is the status of the process (version 6.1.1):


```
oauth2-proxy.service - oauth2-proxy daemon service
     Loaded: loaded (/etc/systemd/system/oauth2-proxy.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Sun 2020-11-15 11:17:12 UTC; 12s ago
   Main PID: 18950 (code=exited, status=1/FAILURE)

Nov 15 11:17:12 localhost.localdomain systemd[1]: oauth2-proxy.service: Main process exited, code=exited, status=1/FAILURE
Nov 15 11:17:12 localhost.localdomain systemd[1]: oauth2-proxy.service: Failed with result 'exit-code'.
Nov 15 11:17:12 localhost.localdomain systemd[1]: oauth2-proxy.service: Scheduled restart job, restart counter is at 5.
Nov 15 11:17:12 localhost.localdomain systemd[1]: Stopped oauth2-proxy daemon service.
Nov 15 11:17:12 localhost.localdomain systemd[1]: oauth2-proxy.service: Start request repeated too quickly.
Nov 15 11:17:12 localhost.localdomain systemd[1]: oauth2-proxy.service: Failed with result 'exit-code'.
Nov 15 11:17:12 localhost.localdomain systemd[1]: Failed to start oauth2-proxy daemon service.
```

**Comment by JoelSpeed** [comment]
Looks like you're on a systemd based system, so try `journalctl -u oauth2-proxy`, that should dump the logs for the service.

One thing I just noticed looking back at this is the flag option you are using `-config`, try `--config` instead with the double dash, we changed the flag parsing library from v5 to v6 which means I think the flags have to be double dash now

**Comment by gianluca1606** [comment]
> journalctl -u oauth2-proxy

I actually also noticed that on the configuration guide --config was mentioned instead of -config, so I already changed to --config. Version 5.1.1 works with both for me --config or -config but 6.1.1 still have the same error.

Here are the logs...
![image](https://user-images.githubusercontent.com/61363342/99184017-645ae200-2740-11eb-96df-f3e83fcd974c.png)

**Comment by gianluca1606** [comment]
I found out the problem, my cookie secret was 20 bytes long but in version 6.0.0 or higher it has to be 24 bytes or 32 bytes. I did not noticed this so I was never able to get oauth2-proxy to run. And another thing that also lead to all this error was, that the oauth2-proxy file in version 5.1.1 was named oauth2_proxy and in the newer versions the name now ist oauth2-proxy. This resulted in me creating system links to a file that never existed. 

Thank you for your effort and for the project. :)
